packages/flutter/lib/src/rendering/sliver_grid.dart

```diff
@@ -606,7 +606,6 @@ class RenderSliverGrid extends RenderSliverMultiBoxAdaptor {
 
     final int lastIndex = indexOf(lastChild!);
 
-    assert(childScrollOffset(firstChild!)! <= scrollOffset);
```

Comment:
I am a bit worrying about removing this assert. If you take a look at line 529, the getMinChildIndexForScrollOffset should return child index whose scroll offset are <= scrollOffset

It feels like a user error to me.

Reply:
after taking a look at the code again it seems fine to remove it. Let me think about it further if there is any corner case

Reply:
I think this is ok